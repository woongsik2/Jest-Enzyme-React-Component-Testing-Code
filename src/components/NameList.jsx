import React, { Component } from 'react';

class NameList extends Component {
    static defultProps = {
        names: []
    }
    renderList(){
        const {names} = this.props;
        const NameList = names.map(
            (name, i) => (<li key={i}>{name}</li>)
        );
        return NameList;
    }
    render() {
        return (
            <ul>
                {this.renderList()}
            </ul>
        );
    }
}

export default NameList;