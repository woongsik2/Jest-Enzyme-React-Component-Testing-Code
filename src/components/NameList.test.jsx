import React from 'react';
import renderer from 'react-test-renderer';
import NameList from './NameList.jsx';

describe('NameList View', () => {
    let component = null;

    it('NameList View 초기 렌더링', () =>{
        component = renderer.create(<NameList names={['웅식', '킴웅식']} />);
    });
    
    it('NameList View Snapshot check', () =>{
        const snap = component.toJSON();
        expect(snap).toMatchSnapshot();
    })

});