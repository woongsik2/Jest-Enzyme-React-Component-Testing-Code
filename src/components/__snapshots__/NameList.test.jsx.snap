// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NameList View NameList View Snapshot check 1`] = `
<ul>
  <li>
    웅식
  </li>
  <li>
    킴웅식
  </li>
</ul>
`;
